# Практичекое задание 1
## по теме: Случайные события. Условная вероятность. Формула Байеса. Независимые испытания

<div style="border:solid lightseagreen 3px; padding: 20px">
<b>Задача 1.</b>
    Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.
</div>

а) Вероятность того, что 1-ая карта - крести:

$$P(А_1) = \frac{13}{52} = \frac{1}{4}$$

Вероятность того, что 2-ая карта - крести:

$$P(А_2) = \frac{12}{51} = \frac{4}{17}$$

Вероятность того, что 3-ая карта - крести:

$$P(А_3) = \frac{11}{50}$$

Вероятность того, что 4-ая карта - крести:

$$P(А_4) = \frac{10}{49}$$

Следовательно, вероятность того, что все 4 карты - крести:

$$P(А_1\:A_2\:A_3\:A_4) = \frac{1}{4} \cdot \frac{4}{17} \cdot \frac{11}{50} \cdot \frac{10}{49} = \frac{11}{4165}$$

б) Вероятность того, что среди 4-рех карт один туз:

$$P(А_1) = \frac{4}{52} = \frac{1}{13}$$

Вероятность того, что среди 4-рех карт два туза:

$$P(A_2) = \frac{4}{52} \cdot \frac{3}{51} = \frac{1}{221}$$

Вероятность того, что среди 4-рех карт три туза:

$$P(A_3) = \frac{4}{52} \cdot \frac{3}{51} \cdot \frac{2}{50} = \frac{1}{5525}$$

Вероятность того, что среди 4-рех карт четыре туза:

$$P(A_4) = \frac{4}{52} \cdot \frac{3}{51} \cdot \frac{2}{50} \cdot \frac{1}{49} = \frac{1}{270725}$$

Вероятность того, что среди 4 карт есть хотя бы 1 туз (1 и больще):

$$P(А_1 + A_2 + A_3 + A_4) = \frac{1}{13} + \frac{1}{221} + \frac{1}{5525} + \frac{1}{270725} = \frac{4}{49}$$

<div style="border:solid lightseagreen 3px; padding: 20px">
<b>Задача 2.</b>
    На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?
</div>

Вероятность равна числу **сочетаний** из n = 10 элементов по k = 3 элементов в каждом.

$$C_n^k = \frac{n!}{k!(n - k)!}$$

In [1]:
from math import factorial

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [3]:
combinations(10, 3)

120

<div style="border:solid lightseagreen 3px; padding: 20px">
<b>Задача 3.</b>
    В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?
</div>

Число всех возможных исходов событий равно числу **сочетаний** из n = 15 элементов по k = 3 элементов:

In [5]:
combinations(15, 3)

455

Число всех благоприятных исходов событий равно числу **сочетаний** из n = 9 элементов по k = 3 элементов:

In [6]:
combinations(9, 3)

84

Вероятность благоприятного события равна:

$$P(А) = \frac{84}{455} = \frac{12}{65}$$

<div style="border:solid lightseagreen 3px; padding: 20px">
<b>Задача 4.</b>
    В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?
</div>

Число всех возможных исходов событий равно числу **сочетаний** из n = 100 элементов по k = 2 элементов:

In [7]:
combinations(100, 2)

4950

Число всех благоприятных исходов событий равно числу **сочетаний** из n = 2 элементов по k = 2 элементов:

In [8]:
combinations(2, 2)

1

Вероятность благоприятного события равна:

$$P(А) = \frac{1}{4950}$$